domain/model/modules.py

```diff
@@ -19,7 +19,7 @@ def __init__(self, hidden_size: int, activation: nn.Module, depth: int) -> None:
         """
         super(MLP_Block, self).__init__()
         layers = []
-        for _ in range(depth):
+        for _ in range(depth-1):
```

Comment:
So if depth is 1, this will return an empty list? I think that is not very intuitive. It may be better to change the depth parameter range to 0-2 instead.

Reply:
There was a bit of a redundancy in the build because on line 68, the network is given a built in default depth of 1. I can create a work-around, but in terms of code-readability, this build made more sense to me.

Reply:
I see. It may be better to pass `depth-1` in line 68 but up to you. Also, there is a mismatch between line 46 and 56. Line 56 should say the network depth default is 2.